src/Orleans/SystemTargetInterfaces/IManagementGrain.cs

```diff
@@ -68,27 +68,55 @@ public interface IManagementGrain : IGrainWithIntegerKey
         /// <param name="hostsIds">List of silos this command is to be sent to.</param>
         /// <returns>Completion promise for this operation.</returns>
         Task<SiloRuntimeStatistics[]> GetRuntimeStatistics(SiloAddress[] hostsIds);
+
         /// <summary>
         /// Return the most recent grain statistics information, amalgomated across silos.
         /// </summary>
         /// <param name="hostsIds">List of silos this command is to be sent to.</param>
         /// <returns>Completion promise for this operation.</returns>
         Task<SimpleGrainStatistic[]> GetSimpleGrainStatistics(SiloAddress[] hostsIds);
+
         /// <summary>
         /// Return the most recent grain statistics information, amalgomated across all silos.
         /// </summary>
         /// <returns>Completion promise for this operation.</returns>
         Task<SimpleGrainStatistic[]> GetSimpleGrainStatistics();
+
         /// <summary>
-        /// Return the most recent activation count for a specific grain across all silos.
+        /// Returns the most recent detailed grain statistics information, amalgomated across silos.
         /// </summary>
-        /// <param name="grainReference">Reference to the grain to be queried.</param>
-        /// <returns>Completion promise for this operation.</returns>
+        /// <param name="hostsIds">List of silos this command is to be sent to.</param>
+        /// <returns></returns>
+        Task<DetailedGrainStatistic[]> GetDetailedGrainStatistics();
+
+        /// <summary>
+        /// Returns the most recent detailed grain statistics information, amalgomated across silos.
+        /// </summary>
+        /// <param name="types">Array of grain types to filter the results with</param>
+        /// <returns></returns>
+        Task<DetailedGrainStatistic[]> GetDetailedGrainStatistics(string[] types);
+
+        /// <summary>
+        /// Returns the most recent detailed grain statistics information, amalgomated across silos.
+        /// </summary>
+        /// <param name="hostsIds">List of silos this command is to be sent to.</param>
+        /// <returns></returns>
+        Task<DetailedGrainStatistic[]> GetDetailedGrainStatistics(SiloAddress[] hostsIds);
+
+        /// <summary>
+        /// Returns the most recent detailed grain statistics information, amalgomated across silos for the specified types.
+        /// </summary>
+        /// <param name="hostsIds">List of silos this command is to be sent to.</param>
+        /// <param name="types">Array of grain types to filter the results with</param>
+        /// <returns></returns>
+        Task<DetailedGrainStatistic[]> GetDetailedGrainStatistics(SiloAddress[] hostsIds, string[] types);
```

Comment:
Wouldn't it be better to have a single method with default arguments?

`GetDetailedGrainStatistics(string[] types = null, SiloAddress[] hostsIds=null)`

where nulls mean all types and all silos respectively?

I would also flip the order of arguments assuming that specifying grain types would be a more common case than specifying a list of silos. That way `GetDetailedGrainStatistics()` and `GetDetailedGrainStatistics(grainTypes)` would be convenient shortcuts.